saleor/checkout/complete_checkout.py

```diff
@@ -1299,10 +1305,18 @@ def complete_checkout(
     # paid is used. In case when we have TRANSACTION_FLOW we use transaction flow to
     # finalize the checkout.
     checkout_is_zero = checkout_info.checkout.total.gross.amount == Decimal(0)
-    if transactions or (
-        checkout_is_zero
-        and checkout_info.channel.order_mark_as_paid_strategy
-        == MarkAsPaidStrategy.TRANSACTION_FLOW
+    if (
```

Comment:
I think that we should have here something like this:
```python
    is_transaction_flow = (
            checkout_info.channel.order_mark_as_paid_strategy
            == MarkAsPaidStrategy.TRANSACTION_FLOW
    )
    use_transaction_flow = (
        # handle with transaction if we have assigned transaction to the checkout
        transactions or
        # handle with transaction if we allow to place the unpaid order from checkout
        checkout_info.channel.allow_to_create_order_without_payment or
        # handle with transaction if we have zero total and we use transaction flow
        # to mark order as paid
        (checkout_is_zero and is_transaction_flow)
    )
    if handle_with_transaction:
        order = complete_checkout_with_transaction(
```
I don't think that we should combine `checkout_info.channel.allow_to_create_order_without_payment` and `checkout_info.channel.order_mark_as_paid_strategy` into `and` condition. It would mean, that you would have to set up 2 different flag to be able to create unpaid order.